com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -504,6 +509,196 @@ internal void InvokeBehaviourOnGainedOwnership()
             }
         }
 
+        private bool m_IsReparented;
+        private ulong? m_LatestParent;
+        private Transform m_CachedParent;
+
+        internal void InvokeBehaviourOnNetworkObjectParentChanged(NetworkObject parentNetworkObject)
+        {
+            for (int i = 0; i < ChildNetworkBehaviours.Count; i++)
+            {
+                ChildNetworkBehaviours[i].OnNetworkObjectParentChanged(parentNetworkObject);
+            }
+        }
+
+        internal void SetCachedParent(Transform parentTransform)
+        {
+            m_CachedParent = parentTransform;
+        }
+
+        private void OnTransformParentChanged()
+        {
+            if (transform.parent == m_CachedParent)
+            {
+                return;
+            }
+
+            if (NetworkManager == null || !NetworkManager.IsListening)
+            {
+                transform.parent = m_CachedParent;
+                Debug.LogException(new NotListeningException($"{nameof(NetworkManager)} is not listening, start a server or host before reparenting"));
+                return;
+            }
+
+            if (!NetworkManager.IsServer)
+            {
+                transform.parent = m_CachedParent;
+                Debug.LogException(new NotServerException($"Only the server can reparent {nameof(NetworkObject)}s"));
+                return;
+            }
+
+            if (!IsSpawned)
+            {
+                transform.parent = m_CachedParent;
+                Debug.LogException(new SpawnStateException($"{nameof(NetworkObject)} can only be reparented after being spawned"));
+                return;
+            }
+
+            var parentTransform = transform.parent;
+            if (parentTransform != null)
+            {
+                var parentObject = transform.parent.GetComponent<NetworkObject>();
+                if (parentObject != null)
+                {
+                    m_LatestParent = parentObject.NetworkObjectId;
+                }
+                else
+                {
+                    if (NetworkLog.CurrentLogLevel <= LogLevel.Error)
```

Comment:
it's not really part of your review, but this if should really be part of NetworkLog.LogError

Reply:
to me, I'd throw away the `NetworkLog` entirely LOL

Reply:
meh I do like the developer level logs. I agree eventually we should have network traces to replace them, but it's a good thing we can enable/disable them.

Reply:
we already have developer-level logs, it's called `Debug.Log` suite and if you were to look at `NetworkLog` closely, it's just a thin layer on top of `Debug.Log` which serves very little value IMHO.